### DISCUSSION TITLE
Group by mono-repo first then minor-patch

### DISCUSSION THREAD

**Discussion by chezsmithy** [renovatebot/renovate#21099] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd like to figure out a way to group minor and patch updates first by mono-repo, and then bucket all the rest of the minor and patch updates into a single group, and finally have all major updates go the dependency dashboard grouped by mono-repo. The allNonMajor preset overrides the mono-repo grouping, so you get a single giant PR for all minor-patch updates.

### What you tried so far.

I've looked at the source code, but can't seem to find a way to create a configuration to group this way. I'm willing to work on an enhancement if this isn't currently possible.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
I think the issue is that any grouping rule you add to config is evaluated _after_ the monorepo grouping rules from presets. To get it to work, you'd need to stop importing `config:base` (which includes the monorepo presets) and instead import each of the presets you want, with the group non-major coming first so that monorepo presets come after and override it.

**Reply by chezsmithy** [reply]
If I put allNonMajor at the top and then add config:base under it would that have the same effect since the mono-repo grouping in config base would override it?

**Reply by rarkins** [reply]
Yes, which is what I thought you're asking. 

It's not so much override but "take precedence" (last matching rule wins)

**Comment by chezsmithy** [comment]
I'd rather suggest that if folks want to group by mono repo first and then major/minor that they import group:allNonMajor first and then config:base. This gives the desired result. 

@HonkingGoose I wonder how we might better explain this in the default presets or documentation.

**Reply by HonkingGoose** [reply]
First off, sorry for not getting back to you in a long time!

I'm rewriting the preset docs we have now with this PR:

- #31358

I'm also asking the maintainers what they think of your suggestion. I'm not sure if it should go in the docs. Explaining difficult workarounds to get "expected behavior" from Renovate feels a bit wrong to me. Maybe there is a better fix for the problem, like:

- Creating a new Renovate preset that matches your expectations for monorepo, then major/minor?
- Adjusting our `config:recommended` preset, if that matches the expectations of most of our users?
